libs/api/domains/air-discount-scheme/src/lib/flight-leg/flight-leg.service.ts

```diff
@@ -38,35 +41,81 @@ export class FlightLegService {
-      })
+  async getThisYearsUserAndRelationsFlightLegs(
+    auth: User,
+  ): Promise<FlightLeg[]> {
+    const flights = await this.getADSWithAuth(auth)
+      .privateFlightUserControllerGetUserAndRelationsFlights()
       .catch((e) => {
         this.handle4xx(e)
       })
 
-    if (!getFlightsResponse) {
+    if (!flights) {
       return []
     }
+
+    const relations: TUser[] = await this.getUserRelations(auth)
     const flightLegs: FlightLeg[] = []
 
-    getFlightsResponse.forEach((flight) => {
-      if (flight?.flightLegs) {
-        for (const flightLeg of flight.flightLegs) {
-          flightLegs.push({
-            ...flightLeg,
-            flight: {
-              bookingDate: flight.bookingDate,
-              id: flight.id,
-            },
-            travel: `${flightLeg.origin} - ${flightLeg.destination}`,
-          })
+    // The expected return value for the graphql layers has some extra properties
+    // We have to maintain circularity as well since the types are circular.
+    // Therefore we do some interesting object surgery
+    for (const flight of flights) {
+      // Not strictly needed but good practice for type safety
+      const relation = relations.find(
+        (relation) => relation.nationalId === flight.nationalId,
+      )
+      if (!relation) {
+        continue
+      }
+
+      // We construct new objects with the expected model properties
+      const constructedUser: FlightUser = {
+        ...relation,
+        name: `${relation.firstName} ${relation.lastName}`,
+      }
+      const constructedFlightLegs: FlightLeg[] = []
+
+      // UserInfo in flight.userInfo has gender as string in the generated schema
+      // but is a string union type, hence the `as Flight` coercion
+      const constructedFlight: Flight = {
+        ...flight,
+        user: constructedUser,
+        flightLegs: [],
+      } as Flight
+
+      // We loop through the flightLegs and attach the extra information needed
+      // as well as attaching a reference to the new constructed flight
+      for (const flightLeg of flight.flightLegs ?? []) {
+        const constructedFlightLeg: FlightLeg = {
+          ...flightLeg,
+          travel: `${flightLeg.origin} - ${flightLeg.destination}`,
+          flight: constructedFlight,
         }
+
+        // Now we attach the flightLeg to its flight reference
+        constructedFlightLeg.flight.flightLegs.push(constructedFlightLeg)
+
+        constructedFlightLegs.push(constructedFlightLeg)
       }
-    })
+      // Add the flightlegs to the return pool
+      flightLegs.push(...constructedFlightLegs)
+    }
 
     return flightLegs
   }
+
+  private async getUserRelations(auth: User): Promise<TUser[]> {
+    const getRelationsResponse = await this.getADSWithAuth(auth)
+      .privateUserControllerGetUserRelations({ nationalId: auth.nationalId })
+      .catch((e) => {
+        this.handle4xx(e)
+      })
+
+    if (!getRelationsResponse) {
+      return []
+    }
```

Comment:
Does this endpoint ever return 404?

Reply:
If memory serves, yes. When `/forsja` doesn't find any wards it returns a 404 instead of an empty array.
I'll double check just in case.

Reply:
Memory didn't serve. It's a 400.
Changed to a simple handleError.